Sources/ParseSwift/Parse Types/GeoPoint.swift

```diff
@@ -1,15 +1,168 @@
 import Foundation
+import CoreLocation
 
-public struct GeoPoint: Codable {
+/**
+  `GeoPoint` may be used to embed a latitude / longitude point as the value for a key in a `ParseObject`.
+   It could be used to perform queries in a geospatial manner using `ParseQuery.-whereKey:nearGeoPoint:`.
+   Currently, instances of `ParseObject` may only have one key associated with a `GeoPoint` type.
+*/
+public struct GeoPoint: Codable, Equatable {
     private let __type: String = "GeoPoint" // swiftlint:disable:this identifier_name
-    public let latitude: Double
-    public let longitude: Double
+    static let earthRadiusMiles = 3958.8
+    static let earthRadiusKilometers = 6371.0
+
+    enum CodingKeys: String, CodingKey {
+        case __type, latitude, longitude // swiftlint:disable:this identifier_name
+    }
+
+    /**
+      Latitude of point in degrees. Valid range is from `-90.0` to `90.0`.
+    */
+    public var latitude: Double {
+        get {
+            return _latitude
+        }
+        set {
+            assert(newValue > -90, "latitude should be > -90")
+            assert(newValue < 90, "latitude should be > 90")
+            _latitude = newValue
+        }
+    }
+
+    /**
+      Longitude of point in degrees. Valid range is from `-180.0` to `180.0`.
+    */
+    public var longitude: Double {
+        get {
+            return _longitude
+        }
+        set {
+            assert(newValue > -180, "longitude should be > -180")
+            assert(newValue < 180, "longitude should be > -180")
+            _longitude = newValue
+        }
+    }
+
+    private var _latitude: Double
+    private var _longitude: Double
+
+    /**
+     Create a GeoPoint instance. Latitude and longitude are set to `0.0`.
+     */
+    public init() {
+        _latitude = 0.0
+        _longitude = 0.0
+    }
+
+    /**
+      Create a new `GeoPoint` instance with the specified latitude and longitude.
+       - parameter latitude: Latitude of point in degrees.
+       - parameter longitude: Longitude of point in degrees.
+     */
     public init(latitude: Double, longitude: Double) {
-        assert(latitude > -180, "latitude should be > -180")
-        assert(latitude < 180, "latitude should be > -180")
+        assert(longitude > -180, "longitude should be > -180")
+        assert(longitude < 180, "longitude should be > -180")
         assert(latitude > -90, "latitude should be > -90")
         assert(latitude < 90, "latitude should be > 90")
-        self.latitude = latitude
-        self.longitude = longitude
+        self._latitude = latitude
+        self._longitude = longitude
+    }
+
+    /**
+      Creates a new `GeoPoint` instance for the given `CLLocation`, set to the location's coordinates.
+       - parameter location: Instace of `CLLocation`, with set latitude and longitude.
+     */
+    public init(location: CLLocation) {
```

Comment:
This should be put in an `available` enclosure or defined another way. I think the best thing is make a protocol called `_CLLocationShim` or something like that and extend `CLLocation` to conform.

Reply:
I'll remove this as well and let the user user just pass long and lat